smtp/server.go

```diff
@@ -28,6 +31,12 @@ func (s *Server) Start() {
 	if err != nil {
 		panic(err.Error())
 	}
+
+	// set a timeout for the inbound server
```

Comment:
this is debatable, accept() doesnt know anything about the server, which is where the timeout should live. the only solution if the timeout is in the server struct is to make accept() part of Server{}'s method set